lib/matplotlib/tests/test_axes.py

```diff
@@ -4713,3 +4713,13 @@ def test_fillbetween_cycle():
         face_target = mcolors.to_rgba('C{}'.format(j))
         assert tuple(cc.get_facecolors().squeeze()) == tuple(face_target)
         assert tuple(cc.get_edgecolors().squeeze()) == tuple(edge_target)
+
+
+@cleanup
+def test_color_length_mismatch():
+    N = 500
+    x, y = np.random.rand(N), np.random.rand(N)
```

Comment:
In the interest of keeping tests minimal and direct, you could reduce N to 5, and use `arange` instead of `rand`.
You could also add a test for the handling of `c` when it is an rgba or sequence of them.